### PR TITLE
Improve French and Chinese localization for figures

### PR DESCRIPTION
**Pull request by MDLC01**
This PR changes the French figure separator from `[ : ]` to `[. -- ]` as it is more common;[^1][^2][^3] and the Chinese figure separator to an em space.[^4]

I also remove some languages from the list of translation. This does not change any behavior because they already used the default `[: ]` separator. As mentioned in https://github.com/typst/typst/pull/2540#issuecomment-1787751837, I should not have added those translations in the first place in #2390.

[^1]: https://theses.hal.science/file/index/docid/346066/filename/Attali.Dominique_1995_these.pdf
[^2]: http://www.numdam.org/item/ITA_1990__24_4_339_0.pdf
[^3]: I asked some people at [ENS Rennes](https://en.wikipedia.org/wiki/%C3%89cole_normale_sup%C3%A9rieure_de_Rennes) and they agreed.
[^4]: https://github.com/typst/typst/pull/2542#issuecomment-1790685258

### CODE REVIEW

**Comment by MDLC01** [comment]
The difference used to be that fr-CH uses a thin no-break space instead of a no-break space.[^1] Now all variantes of french use an en dash.

About the canonical separator for Chinese, it seems to ba a large space (not sure which one, though).[^2][^3][^4]

[^1]: https://fr.wikipedia.org/wiki/Deux-points#Typographie (french)
[^2]: http://gissky.net/paper/UploadFiles_4495/200909/2009091010234287.pdf
[^3]: https://mool.njust.edu.cn/file/up_document/2021/06/_eh6ZazVdiAj1UIp.pdf
[^4]: https://core.ac.uk/download/pdf/84387431.pdf

File: crates/typst-library/src/meta/figure.rs [hunk 1]
```diff
@@ -504,20 +503,11 @@ pub struct FigureCaption {
 impl FigureCaption {
     /// Gets the default separator in the given language and (optionally)
     /// region.
-    fn local_separator(lang: Lang, region: Option<Region>) -> &'static str {
+    fn local_separator(lang: Lang, _region: Option<Region>) -> &'static str {
```
Comment (laurmaedje) [inline crates/typst-library/src/meta/figure.rs:506]:
```suggestion
    fn local_separator(lang: Lang, _: Option<Region>) -> &'static str {
```

**Review by laurmaedje — COMMENTED** [review]
Okay, makes sense. Let's wait for a response from Chinese native speakers for confirmation.

**Comment by Andrew15-5** [comment]
Wait, so English was an exception? Is period used by default (for unlisted languages)? It doesn’t say in the diff.

**Comment by MDLC01** [comment]
I'm not sure I understand what you mean. English uses a colon, French uses an en dash, Chinese seems to use a space. I don't know about other languages, so until we update them, they will use a colon, as English is the default language in Typst (there is a `| _` before the match arm for English).

**Comment by Andrew15-5** [comment]
First of all, merge main into your branch, because my PR merged after your PR was created. In your PR you are removing ˋRUSSIANˋ and since default is using colon, it will effectively make my PR useless and revert Russian back to colon. This is my main concern right now. If default was ˋ.ˋ, then I wouldn't have any issues with that.

**Comment by Enter-tainer** [comment]
Honestly I never notice the separater before. I take a look at [thu thesis](https://github.com/tuna/thuthesis/blob/a32a0f5bc1bf70de6575a9bb3c682bb2ce4dcb1c/thuthesis.dtx#L3583) (LaTeX thesis template of top 1 university in China) and books in my hand. I think it is a quad space.

**Comment by Andrew15-5** [comment]
Thank you.

**Comment by MDLC01** [comment]
From what I have seen, and what people told me, "Fig. _n_. – Légende." seems to be the most common way to caption figures in French (see my original comment). Do you have examples of papers using a colon?

**Comment by Dherse** [comment]
It is most common you're correct, I just meant that if you fully spell it out the, you'd need the colon with spaces around it

**Comment by MDLC01** [comment]
So if I understand what you mean, we should either change the "Figure" text to "Fig." and use the French separator proposed by this PR, or keep the separator that is currently on `main` and "Figure" as the text?

**Comment by Andrew15-5** [comment]
This is an interesting concept. I think 100% of all figures that I've seen in Russian documents use the short form of the supplement. Although it has nothing to do with the previous comments "directly", the idea of making some supplements use the short form (for some languages) is good. We can either just change the default full form to short one or add a "short" boolean. In the second case we would have to store 2 version of supplement for languages that do have short and full forms. This should also affect the default supplement for the ref function. I don't know yet were exactly to put the boolean option.

**Comment by MDLC01** [comment]
Mmh... You are right, maybe we should change some translations to use the abbreviated version when it is more common. My only (minor) fear is that is makes it feel a bit inconsistent (some languages use an abbreviation, others don't). But in the end, the default should be what is most common for a given language.

Not sure changing this is within the scope of this PR, though.

**Comment by Andrew15-5** [comment]
> Not sure changing this is within the scope of this PR, though.

Correct. I was just talking over a small idea, since it is related to this PR.

**Comment by MDLC01** [comment]
To summarize, before merging this, we need to decide whether French should use "Figure _n_ : _légende_" or "Fig. _n_ – légende" (or even "Fig. _n_. – légende"). The former is more consistent with other languages in Typst, while the latter is more common.

**Comment by MDLC01** [comment]
After thinking about it, I believe the default should be whatever is most common for each language. I thus changed the default "Figure" text to "Fig." for French.

I will let @Andrew15-5 make their own PR for Russian translations.

**Comment by MDLC01** [comment]
Changed the title and the description of the PR as the focus shifted toward improving French translations

**Comment by laurmaedje** [comment]
Is this ready to be merged then?

**Comment by MDLC01** [comment]
This PR should now be ready to merge

**Comment by laurmaedje** [comment]
Thanks!

**Pull request merged**